task-canvas/script.js

```diff
@@ -0,0 +1,472 @@
+                this.arrayOfShapes.push(shape);
+                break;
+            case "rectangle":
+                this.clear();
+                shape = new Rectangle(this.ctx, this.color, this.coords.x, this.coords.y, this.coords.vx, this.coords.vy);
+                this.arrayOfShapes.push(shape);
+                break;
+            case "circle":
+                this.clear();
+                shape = new Circle(this.ctx, this.color, this.coords.x, this.coords.y, this.coords.vx, this.coords.vy);
+                this.arrayOfShapes.push(shape);
+                break;
+        }
+        this.arrayOfShapes.forEach(item => item.render(this.ctx));
+    }
+
+    drawShape(x, y, vx, vy, controlType, color) {
+        const ctx = this.ctx;
+        let shape;
+        switch (controlType) {
+            case "new":
+                let deltaX = this.coords.vx - this.coords.x;
+                let deltaY = this.coords.vy - this.coords.y;
+                shape = this.movingShape;
+
+                if (shape) {
+                    shape.moveShape(deltaX, deltaY);
+                    this.coords.x = this.coords.vx;
+                    this.coords.y = this.coords.vy;
+                    this.clear();
+                }
+                break;
+            case "pencil":
+                shape = new BrokenLine(ctx, color, x, y, vx, vy);
+                break;
+            case "rectangle":
+                this.clear();
+                new Rectangle(ctx, color, x, y, vx, vy);
+                break;
+            case "circle":
+                this.clear();
+                new Circle(ctx, color, x, y, vx, vy);
+                break;
+        }
+        this.arrayOfShapes.forEach(item => item.render(ctx));
+    }
+
+    clear() {
+        this.ctx.clearRect(0, 0, canvas.width, canvas.height);
+    }
+
+    getControlType() {
+        let button = document.querySelectorAll('.canvas-checkbox');
+        for (let elem of button) {
+            if (elem.checked) {
+                return `${elem.getAttribute('data-control')}`;
+            };
+        }
+    }
+
+    lineTo(coords, x, y) {
+        coords.push(x, y);
+    }
+
+    mouseMove(event) {
+        let type = this.getControlType;
+        let color = this.color;
+        const ctx = this.ctx;
+
+        let positionBox = event.target.getBoundingClientRect();
+        let canvasX = Math.round(event.clientX - positionBox.left);
+        this.coords.vx = canvasX;
+        let canvasY = Math.round(event.clientY - positionBox.top);
+        this.coords.vy = canvasY;
+            
+        this.drawShape(this.coords.x, this.coords.y, this.coords.vx, this.coords.vy, type(), color);
+        if (type() === 'pencil') {
+            this.lineTo(this.amountCoords, this.coords.vx, this.coords.vy);
+            this.coords.x = this.coords.vx;
+            this.coords.y = this.coords.vy;
+        }
+    }
+}
+
+let goDrow = new CanvasVectorEditor();
```

Comment:
```js
function createCanvasVectorEditor() {
    return new CanvasVectorEditor();
}

createCanvasVectorEditor();
```

I think it would look a little bit better here instead of this line.